src/data/finset/lattice.lean

```diff
@@ -307,6 +307,22 @@ begin
   exact bex_congr (λ b hb, with_bot.coe_lt_coe),
 end
 
+lemma comp_sup'_eq_sup'_comp [semilattice_sup γ] {s : finset β} (H : s.nonempty)
+  {f : β → α} (g : α → γ) (g_sup : ∀ x y, g (x ⊔ y) = g x ⊔ g y) :
+  g (s.sup' H f) = s.sup' H (g ∘ f) :=
+begin
+  rw [←with_bot.coe_eq_coe, coe_sup'],
+  let g' : with_bot α → with_bot γ := with_bot.rec_bot_coe ⊥ (λ x, ↑(g x)),
+  show g' ↑(s.sup' H f) = s.sup (λ a, g' ↑(f a)),
+  rw coe_sup',
+  refine comp_sup_eq_sup_comp g' _ rfl,
+  intros f₁ f₂,
+  cases f₁,
+  { rw [with_bot.none_eq_bot, bot_sup_eq], exact bot_sup_eq.symm, },
+  { cases f₂, refl,
+    exact congr_arg coe (g_sup f₁ f₂), },
```

Comment:
Alternatively, @semorrison's proof from #7249:
```suggestion
  classical,
  revert H,
  apply finset.cons_induction_on s,
  { rintro ⟨a, ⟨⟩⟩, },
  { rintros a s' nm ih n',
    by_cases n : s'.nonempty,
    { rw [finset.sup'_cons n, finset.sup'_cons n, ←ih n, g_sup], },
    { have e : s' = ∅ := finset.not_nonempty_iff_eq_empty.mp n,
      subst e,
      refl, }, },
```
It doesn't save any lines though, and the current proof seems nice because it builds on top of `comp_sup_eq_sup_comp`

Reply:
No problem, yours is fine.